projects/knora-ui/src/lib/viewer/values/date-value/date-input/date-input.component.ts

```diff
@@ -0,0 +1,332 @@
+    this.stateChanges.next();
+  }
+
+  private _required = false;
+
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+    this._disabled ? this.form.disable() : this.form.enable();
+    this.stateChanges.next();
+  }
+
+  private _disabled = false;
+
+  @Input()
+  get placeholder() {
+    return this._placeholder;
+  }
+
+  set placeholder(plh) {
+    this._placeholder = plh;
+    this.stateChanges.next();
+  }
+
+  private _placeholder: string;
+
+  @Input() readonly = false;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  @Input()
+  get value(): KnoraDate | KnoraPeriod | null {
+
+    if (!this.form.valid) {
+      return null;
+    }
+
+    const userInput = this.form.value;
+
+    if (!this.isPeriodControl.value) {
+      // single date
+      if (userInput.dateStart !== null) {
+        return new KnoraDate(userInput.dateStart.calendarName.toUpperCase(), 'CE', userInput.dateStart.calendarStart.year, userInput.dateStart.calendarStart.month, userInput.dateStart.calendarStart.day);
+      } else {
+        return null;
+      }
+    } else {
+      // period
+      if (userInput.dateStart !== null && userInput.dateEnd !== null) {
+
+        const start = new KnoraDate(userInput.dateStart.calendarName.toUpperCase(), 'CE', userInput.dateStart.calendarStart.year, userInput.dateStart.calendarStart.month, userInput.dateStart.calendarStart.day);
+        const end = new KnoraDate(userInput.dateEnd.calendarName.toUpperCase(), 'CE', userInput.dateEnd.calendarStart.year, userInput.dateEnd.calendarStart.month, userInput.dateEnd.calendarStart.day);
+
+        return new KnoraPeriod(start, end);
+      } else {
+        return null;
+      }
+    }
+  }
+
+  set value(date: KnoraDate | KnoraPeriod | null) {
+    if (date !== null) {
+      if (date instanceof KnoraDate) {
+        // single date
+
+        this.form.setValue({
+          dateStart: this.createCalendarDate(date),
+          dateEnd: null,
+          isPeriod: false
+        });
+
+        this.startDateControl.updateValueAndValidity();
+
+      } else {
+        // period
+        const period = date as KnoraPeriod;
+
+        this.form.setValue({
+          dateStart: this.createCalendarDate(period.start),
+          dateEnd: this.createCalendarDate(period.end),
+          isPeriod: true
+        });
+
+        this.startDateControl.updateValueAndValidity();
+
+      }
+    } else {
+      this.form.setValue({dateStart: null, dateEnd: null, isPeriod: false});
+
+      this.startDateControl.updateValueAndValidity();
+    }
+    this.stateChanges.next();
+  }
+
+  /**
+   * Given a `KnoraDate`, creates a Gregorian or Julian calendar date.
+   *
+   * @param date the given KnoraDate.
+   */
+  createCalendarDate(date: KnoraDate): GregorianCalendarDate | JulianCalendarDate {
```

Comment:
What do you think about moving this logic to the js-lib inside a `ConvertUtil` or `CalendarUtil` class? Might be nice to have in case we need to convert from a `KnoraDate` somewhere else.

Reply:
That's an excellent idea. I created https://dasch.myjetbrains.com/youtrack/issue/DSP-139.